Getting a compile error. Code looks good to me otherwise. I can't gameplay test though.

```
FAILED: TR1X.exe.p/src_game_gameflow.c.obj 
/usr/bin/i686-w64-mingw32-gcc -ITR1X.exe.p -I. -I../.. -I../../src -I/ext//include -I/ext/include -I/ext/include/SDL2 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -Wno-unused -D_GNU_SOURCE -DMESON_BUILD -Dmain=SDL_main -MD -MQ TR1X.exe.p/src_game_gameflow.c.obj -MF TR1X.exe.p/src_game_gameflow.c.obj.d -o TR1X.exe.p/src_game_gameflow.c.obj -c ../../src/game/gameflow.c
../../src/game/gameflow.c: In function ‘GameFlow_InterpretSequence’:
../../src/game/gameflow.c:1313:13: error: a label can only be part of a statement and a declaration is not a statement
 1313 |             PHASE_STATS_DATA phase_args = {
      |             ^~~~~~~~~~~~~~~~
```

--- a/meson.build
+++ b/meson.build
@@ -234,6 +234,7 @@ sources = [
   'src/game/phase/phase_game.c',
   'src/game/phase/phase_pause.c',
   'src/game/phase/phase_picture.c',
+  'src/game/phase/phase_stats.c',
   'src/game/picture.c',
   'src/game/random.c',
   'src/game/requester.c',

--- a/src/game/game/game.c
+++ b/src/game/game/game.c
@@ -69,6 +69,8 @@ void Game_SetStatus(GAME_STATUS status)
         Phase_Set(PHASE_PAUSE, NULL);
     } else if (status == GS_IN_GAME) {
         Phase_Set(PHASE_GAME, NULL);
+    } else if (status == GS_IN_STATS) {
+        Phase_Set(PHASE_STATS, NULL);
     } else {
         Phase_Set(PHASE_NULL, NULL);
     }

--- a/src/game/gameflow.c
+++ b/src/game/gameflow.c
@@ -13,7 +13,6 @@
 #include "game/phase/phase_picture.h"
 #include "game/phase/phase_stats.h"
 #include "game/room.h"
-#include "game/stats.h"
 #include "global/const.h"
 #include "global/vars.h"
 #include "json/json_base.h"
@@ -1324,10 +1323,19 @@ GameFlow_InterpretSequence(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
 
         case GFS_TOTAL_STATS:
             if (g_Config.enable_total_stats && level_type != GFL_SAVED) {
-                GAMEFLOW_DISPLAY_PICTURE_DATA *data = seq->data;
-                Stats_ShowTotal(
-                    data->path,
-                    level_type == GFL_BONUS ? GFL_BONUS : GFL_NORMAL);
+                const GAMEFLOW_DISPLAY_PICTURE_DATA *data = seq->data;
+                PHASE_STATS_DATA phase_args = {
+                    .level_num = level_num,
+                    .background_path = data->path,
+                    .total = true,
+                    .level_type =
+                        level_type == GFL_BONUS ? GFL_BONUS : GFL_NORMAL,
+                };
+                Phase_Set(PHASE_STATS, &phase_args);
+                ret = Game_Loop();
+                if (ret != GF_NOP) {
+                    return ret;
+                }
             }
             break;
 

--- a/src/game/gameflow.c
+++ b/src/game/gameflow.c
@@ -11,6 +11,7 @@
 #include "game/objects/creatures/bacon_lara.h"
 #include "game/phase/phase.h"
 #include "game/phase/phase_picture.h"
+#include "game/phase/phase_stats.h"
 #include "game/room.h"
 #include "game/stats.h"
 #include "global/const.h"
@@ -1309,9 +1310,17 @@ GameFlow_InterpretSequence(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
             }
             break;
 
-        case GFS_LEVEL_STATS:
-            Stats_Show((int32_t)(intptr_t)seq->data);
+        case GFS_LEVEL_STATS: {
+            PHASE_STATS_DATA phase_args = {
+                .level_num = (int32_t)(intptr_t)seq->data,
+            };
+            Phase_Set(PHASE_STATS, &phase_args);
+            ret = Game_Loop();
+            if (ret != GF_NOP) {
+                return ret;
+            }
             break;
+        }
 
         case GFS_TOTAL_STATS:
             if (g_Config.enable_total_stats && level_type != GFL_SAVED) {
@@ -1337,7 +1346,10 @@ GameFlow_InterpretSequence(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
                 .display_time = data->display_time,
             };
             Phase_Set(PHASE_PICTURE, &phase_arg);
-            Game_Loop();
+            ret = Game_Loop();
+            if (ret != GF_NOP) {
+                return ret;
+            }
             break;
 
         case GFS_EXIT_TO_TITLE:

--- a/src/game/phase/phase.c
+++ b/src/game/phase/phase.c
@@ -3,6 +3,7 @@
 #include "game/phase/phase_game.h"
 #include "game/phase/phase_pause.h"
 #include "game/phase/phase_picture.h"
+#include "game/phase/phase_stats.h"
 #include "global/types.h"
 #include "log.h"
 
@@ -32,6 +33,10 @@ void Phase_Set(const PHASE phase, void *arg)
 
     case PHASE_PICTURE:
         m_Phaser = &g_PicturePhaser;
+        break;
+
+    case PHASE_STATS:
+        m_Phaser = &g_StatsPhaser;
         break;
     }
 

--- a/src/game/phase/phase.h
+++ b/src/game/phase/phase.h
@@ -9,6 +9,7 @@ typedef enum PHASE {
     PHASE_GAME,
     PHASE_PAUSE,
     PHASE_PICTURE,
+    PHASE_STATS,
 } PHASE;
 
 typedef struct PHASER {

--- a/src/game/phase/phase_picture.c
+++ b/src/game/phase/phase_picture.c
@@ -19,6 +19,11 @@ static PICTURE_STATE m_State = PICTURE_STATE_FADE_IN;
 static int32_t m_StartTime = 0;
 static double m_DisplayTime = 0.0;
 
+static void Phase_Picture_Start(void *arg);
+static void Phase_Picture_End(void);
+static GAMEFLOW_OPTION Phase_Picture_Control(int32_t nframes);
+static void Phase_Picture_Draw(void);
+
 static void Phase_Picture_Start(void *arg)
 {
     const PHASE_PICTURE_DATA *data = (const PHASE_PICTURE_DATA *)arg;
@@ -36,7 +41,7 @@ static void Phase_Picture_End(void)
     Output_FadeReset();
 }
 
-static int32_t Phase_Picture_Control(int32_t nframes)
+static GAMEFLOW_OPTION Phase_Picture_Control(int32_t nframes)
 {
     Input_Update();
     Shell_ProcessInput();
@@ -53,20 +58,15 @@ static int32_t Phase_Picture_Control(int32_t nframes)
         break;
 
     case PICTURE_STATE_DISPLAY:
-        if (Clock_GetMS() - m_StartTime >= m_DisplayTime * 1000) {
-            m_State = PICTURE_STATE_FADE_OUT;
-        }
-        if (g_InputDB.any) {
+        if (g_InputDB.any
+            || (Clock_GetMS() - m_StartTime >= m_DisplayTime * 1000)) {
             m_State = PICTURE_STATE_FADE_OUT;
         }
         break;
 
     case PICTURE_STATE_FADE_OUT:
         Output_FadeToBlack(true);
-        if (!Output_FadeIsAnimating()) {
-            return GF_NOP_BREAK;
-        }
-        if (g_InputDB.any) {
+        if (g_InputDB.any || !Output_FadeIsAnimating()) {
             return GF_NOP_BREAK;
         }
         break;

--- a/src/game/phase/phase_stats.c
+++ b/src/game/phase/phase_stats.c
@@ -293,9 +293,10 @@ static GAMEFLOW_OPTION Phase_Stats_Control(int32_t nframes)
 
     switch (m_State) {
     case STATS_STATE_FADE_IN:
-        if (!Output_FadeIsAnimating() || g_InputDB.menu_confirm
-            || g_InputDB.menu_back) {
+        if (!Output_FadeIsAnimating()) {
             m_State = STATS_STATE_DISPLAY;
+        } else if (g_InputDB.menu_confirm || g_InputDB.menu_back) {
+            m_State = STATS_STATE_FADE_OUT;
         }
         break;
 

--- a/src/game/phase/phase_stats.c
+++ b/src/game/phase/phase_stats.c
@@ -1,0 +1,207 @@
+#include "game/phase/phase_stats.h"
+
+#include "config.h"
+#include "game/game.h"
+#include "game/gameflow.h"
+#include "game/input.h"
+#include "game/output.h"
+#include "game/overlay.h"
+#include "game/phase/phase.h"
+#include "game/shell.h"
+#include "game/text.h"
+#include "global/const.h"
+#include "global/types.h"
+#include "global/vars.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#define MAX_TEXTSTRINGS 10
+
+typedef enum STATS_STATE {
+    STATS_STATE_FADE_IN,
+    STATS_STATE_DISPLAY,
+    STATS_STATE_FADE_OUT,
+} STATS_STATE;
+
+static GAME_STATUS m_OldGameStatus;
+static STATS_STATE m_State = STATS_STATE_DISPLAY;
+static TEXTSTRING *m_Texts[MAX_TEXTSTRINGS] = { 0 };
+
+static void Phase_Stats_CreateTexts(int32_t level_num);
+static void Phase_Stats_Start(void *arg);
+static void Phase_Stats_End(void);
+static GAMEFLOW_OPTION Phase_Stats_Control(int32_t nframes);
+static void Phase_Stats_Draw(void);
+
+static void Phase_Stats_CreateTexts(int32_t level_num)
+{
+    char buf[100];
+    char time_str[100];
+
+    const GAME_STATS *stats = &g_GameInfo.current[level_num].stats;
+
+    Overlay_HideGameInfo();
+
+    int y = -50;
+    const int row_height = 30;
+
+    TEXTSTRING **cur_txt = &m_Texts[0];
+
+    // heading
+    sprintf(buf, "%s", g_GameFlow.levels[level_num].level_title);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // kills
+    sprintf(
+        buf,
+        g_GameFlow.strings
+            [g_Config.enable_detailed_stats ? GS_STATS_KILLS_DETAIL_FMT
+                                            : GS_STATS_KILLS_BASIC_FMT],
+        stats->kill_count, stats->max_kill_count);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // pickups
+    sprintf(
+        buf,
+        g_GameFlow.strings
+            [g_Config.enable_detailed_stats ? GS_STATS_PICKUPS_DETAIL_FMT
+                                            : GS_STATS_PICKUPS_BASIC_FMT],
+        stats->pickup_count, stats->max_pickup_count);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // secrets
+    int secret_count = 0;
+    int16_t secret_flags = stats->secret_flags;
+    for (int i = 0; i < MAX_SECRETS; i++) {
+        if (secret_flags & 1) {
+            secret_count++;
+        }
+        secret_flags >>= 1;
+    }
+    sprintf(
+        buf, g_GameFlow.strings[GS_STATS_SECRETS_FMT], secret_count,
+        stats->max_secret_count);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // deaths
+    if (g_Config.enable_deaths_counter && g_GameInfo.death_counter_supported) {
+        sprintf(
+            buf, g_GameFlow.strings[GS_STATS_DEATHS_FMT], stats->death_count);
+        *cur_txt = Text_Create(0, y, buf);
+        Text_CentreH(*cur_txt, 1);
+        Text_CentreV(*cur_txt, 1);
+        cur_txt++;
+        y += row_height;
+    }
+
+    // time taken
+    int seconds = stats->timer / 30;
+    int hours = seconds / 3600;
+    int minutes = (seconds / 60) % 60;
+    seconds %= 60;
+    if (hours) {
+        sprintf(
+            time_str, "%d:%d%d:%d%d", hours, minutes / 10, minutes % 10,
+            seconds / 10, seconds % 10);
+    } else {
+        sprintf(time_str, "%d:%d%d", minutes, seconds / 10, seconds % 10);
+    }
+    sprintf(buf, g_GameFlow.strings[GS_STATS_TIME_TAKEN_FMT], time_str);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+}
+
+static void Phase_Stats_Start(void *arg)
+{
+    const PHASE_STATS_DATA *data = (const PHASE_STATS_DATA *)arg;
+    if (g_CurrentLevel == g_GameFlow.gym_level_num) {
+        Output_FadeToBlack(false);
+        m_State = STATS_STATE_FADE_OUT;
+        return;
+    }
+
+    m_State = STATS_STATE_FADE_IN;
+    m_OldGameStatus = Game_GetStatus();
+    Output_FadeToSemiBlack(true);
+
+    Phase_Stats_CreateTexts(
+        data && data->level_num != -1 ? data->level_num : g_CurrentLevel);
+}
+
+static void Phase_Stats_End(void)
+{
+    Output_FadeReset();
+
+    for (int i = 0; i < MAX_TEXTSTRINGS; i++) {
+        TEXTSTRING **cur_txt = &m_Texts[i];
+        if (*cur_txt) {
+            Text_Remove(*cur_txt);
+        }
+    }
+}
+
+static GAMEFLOW_OPTION Phase_Stats_Control(int32_t nframes)
+{
+    Input_Update();
+    Shell_ProcessInput();
+
+    switch (m_State) {
+    case STATS_STATE_FADE_IN:
+        if (!Output_FadeIsAnimating() || g_InputDB.menu_confirm
+            || g_InputDB.menu_back) {
+            m_State = STATS_STATE_DISPLAY;
+        }
+        break;
+
+    case STATS_STATE_DISPLAY:
+        if (g_InputDB.menu_confirm || g_InputDB.menu_back) {
+            m_State = STATS_STATE_FADE_OUT;
+        }
+        break;
+
+    case STATS_STATE_FADE_OUT:
+        Output_FadeToBlack(true);
+        if (!Output_FadeIsAnimating() || g_InputDB.menu_confirm
+            || g_InputDB.menu_back) {
+            Game_SetStatus(m_OldGameStatus);
+            return GF_NOP_BREAK;
+        }
+        break;
+    }
+
+    return GF_NOP;
+}
+
+static void Phase_Stats_Draw(void)
+{
+    Game_DrawScene(false);
+    Text_Draw();
+}
+
+PHASER g_StatsPhaser = {
+    .start = Phase_Stats_Start,
+    .end = Phase_Stats_End,
+    .control = Phase_Stats_Control,
+    .draw = Phase_Stats_Draw,
+};

--- a/src/game/phase/phase_stats.c
+++ b/src/game/phase/phase_stats.c
@@ -4,15 +4,18 @@
 #include "game/game.h"
 #include "game/gameflow.h"
 #include "game/input.h"
+#include "game/music.h"
 #include "game/output.h"
 #include "game/overlay.h"
 #include "game/phase/phase.h"
 #include "game/shell.h"
+#include "game/stats.h"
 #include "game/text.h"
 #include "global/const.h"
 #include "global/types.h"
 #include "global/vars.h"
 
+#include <assert.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdio.h>
@@ -26,6 +29,7 @@ typedef enum STATS_STATE {
 } STATS_STATE;
 
 static GAME_STATUS m_OldGameStatus;
+static bool m_Total = false;
 static STATS_STATE m_State = STATS_STATE_DISPLAY;
 static TEXTSTRING *m_Texts[MAX_TEXTSTRINGS] = { 0 };
 
@@ -132,9 +136,121 @@ static void Phase_Stats_CreateTexts(int32_t level_num)
     y += row_height;
 }
 
+static void Phase_Stats_CreateTextsTotal(GAMEFLOW_LEVEL_TYPE level_type)
+{
+    TOTAL_STATS stats;
+    Stats_ComputeTotal(level_type, &stats);
+
+    char buf[100];
+    char time_str[100];
+    TEXTSTRING **cur_txt = &m_Texts[0];
+
+    int top_y = 55;
+    int y = 55;
+    const int row_width = 220;
+    const int row_height = 20;
+    int16_t border = 4;
+
+    // reserve space for heading
+    y += row_height + border * 2;
+
+    // kills
+    sprintf(
+        buf,
+        g_GameFlow.strings
+            [g_Config.enable_detailed_stats ? GS_STATS_KILLS_DETAIL_FMT
+                                            : GS_STATS_KILLS_BASIC_FMT],
+        stats.player_kill_count, stats.total_kill_count);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // pickups
+    sprintf(
+        buf,
+        g_GameFlow.strings
+            [g_Config.enable_detailed_stats ? GS_STATS_PICKUPS_DETAIL_FMT
+                                            : GS_STATS_PICKUPS_BASIC_FMT],
+        stats.player_pickup_count, stats.total_pickup_count);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // secrets
+    sprintf(
+        buf, g_GameFlow.strings[GS_STATS_SECRETS_FMT],
+        stats.player_secret_count, stats.total_secret_count);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // deaths
+    if (g_Config.enable_deaths_counter && g_GameInfo.death_counter_supported) {
+        sprintf(
+            buf, g_GameFlow.strings[GS_STATS_DEATHS_FMT], stats.death_count);
+        *cur_txt = Text_Create(0, y, buf);
+        Text_CentreH(*cur_txt, 1);
+        Text_CentreV(*cur_txt, 1);
+        cur_txt++;
+        y += row_height;
+    }
+
+    // time taken
+    int seconds = stats.timer / 30;
+    int hours = seconds / 3600;
+    int minutes = (seconds / 60) % 60;
+    seconds %= 60;
+    if (hours) {
+        sprintf(
+            time_str, "%d:%d%d:%d%d", hours, minutes / 10, minutes % 10,
+            seconds / 10, seconds % 10);
+    } else {
+        sprintf(time_str, "%d:%d%d", minutes, seconds / 10, seconds % 10);
+    }
+    sprintf(buf, g_GameFlow.strings[GS_STATS_TIME_TAKEN_FMT], time_str);
+    *cur_txt = Text_Create(0, y, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    cur_txt++;
+    y += row_height;
+
+    // border
+    int16_t height = y + border * 2 - top_y;
+    *cur_txt = Text_Create(0, top_y, " ");
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    Text_AddBackground(*cur_txt, row_width, height, 0, 0, TS_BACKGROUND);
+    Text_AddOutline(*cur_txt, true, TS_BACKGROUND);
+    cur_txt++;
+
+    // heading
+    sprintf(
+        buf, "%s",
+        g_GameFlow.strings
+            [level_type == GFL_BONUS ? GS_STATS_BONUS_STATISTICS
+                                     : GS_STATS_FINAL_STATISTICS]);
+    *cur_txt = Text_Create(0, top_y + 2, buf);
+    Text_CentreH(*cur_txt, 1);
+    Text_CentreV(*cur_txt, 1);
+    Text_AddBackground(*cur_txt, row_width - 4, 0, 0, 0, TS_HEADING);
+    Text_AddOutline(*cur_txt, true, TS_HEADING);
+    cur_txt++;
+}
+
 static void Phase_Stats_Start(void *arg)
 {
     const PHASE_STATS_DATA *data = (const PHASE_STATS_DATA *)arg;
+    if (data && data->total) {
+        assert(data->level_type);
+        Output_LoadBackdropImage(data->background_path);
+    }
+
     if (g_CurrentLevel == g_GameFlow.gym_level_num) {
         Output_FadeToBlack(false);
         m_State = STATS_STATE_FADE_OUT;
@@ -143,15 +259,24 @@ static void Phase_Stats_Start(void *arg)
 
     m_State = STATS_STATE_FADE_IN;
     m_OldGameStatus = Game_GetStatus();
-    Output_FadeToSemiBlack(true);
+    m_Total = data && data->total;
 
-    Phase_Stats_CreateTexts(
-        data && data->level_num != -1 ? data->level_num : g_CurrentLevel);
+    if (data && data->total) {
+        Phase_Stats_CreateTextsTotal(data->level_type);
+        Output_FadeReset();
+        Output_FadeResetToBlack();
+        Output_FadeToTransparent(true);
+    } else {
+        Phase_Stats_CreateTexts(
+            data && data->level_num != -1 ? data->level_num : g_CurrentLevel);
+        Output_FadeToSemiBlack(true);
+    }
 }
 
 static void Phase_Stats_End(void)
 {
     Output_FadeReset();
+    Music_Stop();
 
     for (int i = 0; i < MAX_TEXTSTRINGS; i++) {
         TEXTSTRING **cur_txt = &m_Texts[i];
@@ -195,7 +320,11 @@ static GAMEFLOW_OPTION Phase_Stats_Control(int32_t nframes)
 
 static void Phase_Stats_Draw(void)
 {
-    Game_DrawScene(false);
+    if (m_Total) {
+        Output_DrawBackdropImage();
+    } else {
+        Game_DrawScene(false);
+    }
     Text_Draw();
 }
 

--- a/src/game/phase/phase_stats.h
+++ b/src/game/phase/phase_stats.h
@@ -1,0 +1,12 @@
+#pragma once
+
+#include "game/phase/phase.h"
+#include "global/types.h"
+
+#include <stdint.h>
+
+typedef struct PHASE_STATS_DATA {
+    int32_t level_num;
+} PHASE_STATS_DATA;
+
+extern PHASER g_StatsPhaser;

--- a/src/game/phase/phase_stats.h
+++ b/src/game/phase/phase_stats.h
@@ -3,10 +3,14 @@
 #include "game/phase/phase.h"
 #include "global/types.h"
 
+#include <stdbool.h>
 #include <stdint.h>
 
 typedef struct PHASE_STATS_DATA {
     int32_t level_num;
+    const char *background_path;
+    bool total;
+    GAMEFLOW_LEVEL_TYPE level_type;
 } PHASE_STATS_DATA;
 
 extern PHASER g_StatsPhaser;

--- a/src/game/stats.c
+++ b/src/game/stats.c
@@ -1,15 +1,9 @@
 #include "game/stats.h"
 
-#include "config.h"
 #include "game/carrier.h"
 #include "game/gamebuf.h"
 #include "game/gameflow.h"
-#include "game/input.h"
 #include "game/items.h"
-#include "game/music.h"
-#include "game/output.h"
-#include "game/shell.h"
-#include "game/text.h"
 #include "global/const.h"
 #include "global/types.h"
 #include "global/vars.h"
@@ -21,18 +15,6 @@
 
 #define MAX_TEXTSTRINGS 10
 
-typedef struct TOTAL_STATS {
-    uint32_t timer;
-    uint32_t death_count;
-    uint32_t player_kill_count;
-    uint16_t player_secret_count;
-    uint16_t player_pickup_count;
-    uint32_t total_kill_count;
-    uint16_t total_secret_count;
-    uint16_t total_pickup_count;
-} TOTAL_STATS;
-
-static TOTAL_STATS m_TotalStats;
 static int32_t m_CachedItemCount = 0;
 static FLOOR_INFO **m_CachedFloorArray = NULL;
 static int32_t m_LevelPickups = 0;
@@ -41,7 +23,6 @@ static int32_t m_LevelSecrets = 0;
 static uint32_t m_SecretRoom = 0;
 static bool m_KillableItems[MAX_ITEMS] = { 0 };
 static bool m_IfKillable[O_NUMBER_OF] = { 0 };
-static void Stats_ComputeTotal(GAMEFLOW_LEVEL_TYPE level_type);
 
 int16_t m_PickupObjs[] = { O_PICKUP_ITEM1,   O_PICKUP_ITEM2,  O_KEY_ITEM1,
                            O_KEY_ITEM2,      O_KEY_ITEM3,     O_KEY_ITEM4,
@@ -188,9 +169,10 @@ static void Stats_IncludeKillableItem(int16_t item_num)
     m_LevelPickups += Carrier_GetItemCount(item_num);
 }
 
-static void Stats_ComputeTotal(GAMEFLOW_LEVEL_TYPE level_type)
+void Stats_ComputeTotal(
+    GAMEFLOW_LEVEL_TYPE level_type, TOTAL_STATS *total_stats)
 {
-    memset(&m_TotalStats, 0, sizeof(TOTAL_STATS));
+    memset(total_stats, 0, sizeof(TOTAL_STATS));
 
     int16_t secret_flags = 0;
 
@@ -198,23 +180,23 @@ static void Stats_ComputeTotal(GAMEFLOW_LEVEL_TYPE level_type)
         if (g_GameFlow.levels[i].level_type != level_type) {
             continue;
         }
-        const GAME_STATS *stats = &g_GameInfo.current[i].stats;
+        const GAME_STATS *level_stats = &g_GameInfo.current[i].stats;
 
-        m_TotalStats.player_kill_count += stats->kill_count;
-        m_TotalStats.player_pickup_count += stats->pickup_count;
-        secret_flags = stats->secret_flags;
+        total_stats->player_kill_count += level_stats->kill_count;
+        total_stats->player_pickup_count += level_stats->pickup_count;
+        secret_flags = level_stats->secret_flags;
         for (int j = 0; j < MAX_SECRETS; j++) {
             if (secret_flags & 1) {
-                m_TotalStats.player_secret_count++;
+                total_stats->player_secret_count++;
             }
             secret_flags >>= 1;
         }
 
-        m_TotalStats.timer += stats->timer;
-        m_TotalStats.death_count += stats->death_count;
-        m_TotalStats.total_kill_count += stats->max_kill_count;
-        m_TotalStats.total_secret_count += stats->max_secret_count;
-        m_TotalStats.total_pickup_count += stats->max_pickup_count;
+        total_stats->timer += level_stats->timer;
+        total_stats->death_count += level_stats->death_count;
+        total_stats->total_kill_count += level_stats->max_kill_count;
+        total_stats->total_secret_count += level_stats->max_secret_count;
+        total_stats->total_pickup_count += level_stats->max_pickup_count;
     }
 }
 
@@ -292,153 +274,9 @@ int32_t Stats_GetSecrets(void)
     return m_LevelSecrets;
 }
 
-void Stats_ShowTotal(const char *filename, GAMEFLOW_LEVEL_TYPE level_type)
-{
-    Stats_ComputeTotal(level_type);
-
-    char buf[100];
-    char time_str[100];
-    TEXTSTRING *all_txt[MAX_TEXTSTRINGS] = { 0 };
-    TEXTSTRING **cur_txt = &all_txt[0];
-
-    int top_y = 55;
-    int y = 55;
-    const int row_width = 220;
-    const int row_height = 20;
-    int16_t border = 4;
-
-    // reserve space for heading
-    y += row_height + border * 2;
-
-    // kills
-    sprintf(
-        buf,
-        g_GameFlow.strings
-            [g_Config.enable_detailed_stats ? GS_STATS_KILLS_DETAIL_FMT
-                                            : GS_STATS_KILLS_BASIC_FMT],
-        m_TotalStats.player_kill_count, m_TotalStats.total_kill_count);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // pickups
-    sprintf(
-        buf,
-        g_GameFlow.strings
-            [g_Config.enable_detailed_stats ? GS_STATS_PICKUPS_DETAIL_FMT
-                                            : GS_STATS_PICKUPS_BASIC_FMT],
-        m_TotalStats.player_pickup_count, m_TotalStats.total_pickup_count);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // secrets
-    sprintf(
-        buf, g_GameFlow.strings[GS_STATS_SECRETS_FMT],
-        m_TotalStats.player_secret_count, m_TotalStats.total_secret_count);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // deaths
-    if (g_Config.enable_deaths_counter && g_GameInfo.death_counter_supported) {
-        sprintf(
-            buf, g_GameFlow.strings[GS_STATS_DEATHS_FMT],
-            m_TotalStats.death_count);
-        *cur_txt = Text_Create(0, y, buf);
-        Text_CentreH(*cur_txt, 1);
-        Text_CentreV(*cur_txt, 1);
-        cur_txt++;
-        y += row_height;
-    }
-
-    // time taken
-    int seconds = m_TotalStats.timer / 30;
-    int hours = seconds / 3600;
-    int minutes = (seconds / 60) % 60;
-    seconds %= 60;
-    if (hours) {
-        sprintf(
-            time_str, "%d:%d%d:%d%d", hours, minutes / 10, minutes % 10,
-            seconds / 10, seconds % 10);
-    } else {
-        sprintf(time_str, "%d:%d%d", minutes, seconds / 10, seconds % 10);
-    }
-    sprintf(buf, g_GameFlow.strings[GS_STATS_TIME_TAKEN_FMT], time_str);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // border
-    int16_t height = y + border * 2 - top_y;
-    *cur_txt = Text_Create(0, top_y, " ");
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    Text_AddBackground(*cur_txt, row_width, height, 0, 0, TS_BACKGROUND);
-    Text_AddOutline(*cur_txt, true, TS_BACKGROUND);
-    cur_txt++;
-
-    // heading
-    sprintf(
-        buf, "%s",
-        g_GameFlow.strings
-            [level_type == GFL_BONUS ? GS_STATS_BONUS_STATISTICS
-                                     : GS_STATS_FINAL_STATISTICS]);
-    *cur_txt = Text_Create(0, top_y + 2, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    Text_AddBackground(*cur_txt, row_width - 4, 0, 0, 0, TS_HEADING);
-    Text_AddOutline(*cur_txt, true, TS_HEADING);
-    cur_txt++;
-
-    Output_LoadBackdropImage(filename);
-    Output_FadeReset();
-    Output_FadeResetToBlack();
-    Output_FadeToTransparent(true);
-    while (Output_FadeIsAnimating()) {
-        Output_DrawBackdropImage();
-        Output_DumpScreen();
-    }
-
-    // wait till a skip key is pressed
-    do {
-        Output_DrawBackdropImage();
-
-        Input_Update();
-        Shell_ProcessInput();
-
-        Text_Draw();
-        Output_DumpScreen();
-    } while (!g_InputDB.menu_confirm && !g_InputDB.menu_back);
-
-    // fade out
-    Output_FadeToBlack(true);
-    while (Output_FadeIsAnimating()) {
-        Output_DrawBackdropImage();
-        Output_DumpScreen();
-    }
-
-    Output_FadeReset();
-    Music_Stop();
-
-    for (int i = 0; i < MAX_TEXTSTRINGS; i++) {
-        cur_txt = &all_txt[i];
-        if (*cur_txt) {
-            Text_Remove(*cur_txt);
-        }
-    }
-}
-
 bool Stats_CheckAllSecretsCollected(GAMEFLOW_LEVEL_TYPE level_type)
 {
-    Stats_ComputeTotal(level_type);
-    return m_TotalStats.player_secret_count >= m_TotalStats.total_secret_count;
+    TOTAL_STATS total_stats = { 0 };
+    Stats_ComputeTotal(level_type, &total_stats);
+    return total_stats.player_secret_count >= total_stats.total_secret_count;
 }

--- a/src/game/stats.c
+++ b/src/game/stats.c
@@ -2,14 +2,12 @@
 
 #include "config.h"
 #include "game/carrier.h"
-#include "game/game.h"
 #include "game/gamebuf.h"
 #include "game/gameflow.h"
 #include "game/input.h"
 #include "game/items.h"
 #include "game/music.h"
 #include "game/output.h"
-#include "game/overlay.h"
 #include "game/shell.h"
 #include "game/text.h"
 #include "global/const.h"
@@ -292,146 +290,6 @@ int32_t Stats_GetKillables(void)
 int32_t Stats_GetSecrets(void)
 {
     return m_LevelSecrets;
-}
-
-void Stats_Show(int32_t level_num)
-{
-    if (g_CurrentLevel == g_GameFlow.gym_level_num) {
-        Output_FadeToBlack(false);
-        while (Output_FadeIsAnimating()) {
-            Game_DrawScene(false);
-            Output_DumpScreen();
-        }
-        Output_FadeReset();
-        return;
-    }
-
-    const GAME_STATUS old_status = Game_GetStatus();
-    Game_SetStatus(GS_IN_STATS);
-
-    char buf[100];
-    char time_str[100];
-    TEXTSTRING *all_txt[MAX_TEXTSTRINGS] = { 0 };
-    TEXTSTRING **cur_txt = &all_txt[0];
-
-    const GAME_STATS *stats = &g_GameInfo.current[level_num].stats;
-
-    Overlay_HideGameInfo();
-
-    int y = -50;
-    const int row_height = 30;
-
-    // heading
-    sprintf(buf, "%s", g_GameFlow.levels[level_num].level_title);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // kills
-    sprintf(
-        buf,
-        g_GameFlow.strings
-            [g_Config.enable_detailed_stats ? GS_STATS_KILLS_DETAIL_FMT
-                                            : GS_STATS_KILLS_BASIC_FMT],
-        stats->kill_count, stats->max_kill_count);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // pickups
-    sprintf(
-        buf,
-        g_GameFlow.strings
-            [g_Config.enable_detailed_stats ? GS_STATS_PICKUPS_DETAIL_FMT
-                                            : GS_STATS_PICKUPS_BASIC_FMT],
-        stats->pickup_count, stats->max_pickup_count);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // secrets
-    int secret_count = 0;
-    int16_t secret_flags = stats->secret_flags;
-    for (int i = 0; i < MAX_SECRETS; i++) {
-        if (secret_flags & 1) {
-            secret_count++;
-        }
-        secret_flags >>= 1;
-    }
-    sprintf(
-        buf, g_GameFlow.strings[GS_STATS_SECRETS_FMT], secret_count,
-        stats->max_secret_count);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    // deaths
-    if (g_Config.enable_deaths_counter && g_GameInfo.death_counter_supported) {
-        sprintf(
-            buf, g_GameFlow.strings[GS_STATS_DEATHS_FMT], stats->death_count);
-        *cur_txt = Text_Create(0, y, buf);
-        Text_CentreH(*cur_txt, 1);
-        Text_CentreV(*cur_txt, 1);
-        cur_txt++;
-        y += row_height;
-    }
-
-    // time taken
-    int seconds = stats->timer / 30;
-    int hours = seconds / 3600;
-    int minutes = (seconds / 60) % 60;
-    seconds %= 60;
-    if (hours) {
-        sprintf(
-            time_str, "%d:%d%d:%d%d", hours, minutes / 10, minutes % 10,
-            seconds / 10, seconds % 10);
-    } else {
-        sprintf(time_str, "%d:%d%d", minutes, seconds / 10, seconds % 10);
-    }
-    sprintf(buf, g_GameFlow.strings[GS_STATS_TIME_TAKEN_FMT], time_str);
-    *cur_txt = Text_Create(0, y, buf);
-    Text_CentreH(*cur_txt, 1);
-    Text_CentreV(*cur_txt, 1);
-    cur_txt++;
-    y += row_height;
-
-    Output_FadeToSemiBlack(true);
-    // wait till a skip key is pressed
-    do {
-        Game_DrawScene(false);
-
-        Input_Update();
-        Shell_ProcessInput();
-
-        Text_Draw();
-        Output_DumpScreen();
-    } while (!g_InputDB.menu_confirm && !g_InputDB.menu_back);
-
-    Output_FadeToBlack(false);
-
-    // finish fading
-    while (Output_FadeIsAnimating()) {
-        Game_DrawScene(false);
-        Output_DumpScreen();
-    }
-
-    Output_FadeReset();
-    Game_SetStatus(old_status);
-
-    for (int i = 0; i < MAX_TEXTSTRINGS; i++) {
-        cur_txt = &all_txt[i];
-        if (*cur_txt) {
-            Text_Remove(*cur_txt);
-        }
-    }
 }
 
 void Stats_ShowTotal(const char *filename, GAMEFLOW_LEVEL_TYPE level_type)

--- a/src/game/stats.h
+++ b/src/game/stats.h
@@ -5,11 +5,22 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+typedef struct TOTAL_STATS {
+    uint32_t timer;
+    uint32_t death_count;
+    uint32_t player_kill_count;
+    uint16_t player_secret_count;
+    uint16_t player_pickup_count;
+    uint32_t total_kill_count;
+    uint16_t total_secret_count;
+    uint16_t total_pickup_count;
+} TOTAL_STATS;
+
 void Stats_ObserveRoomsLoad(void);
 void Stats_ObserveItemsLoad(void);
 void Stats_CalculateStats(void);
 int32_t Stats_GetPickups(void);
 int32_t Stats_GetKillables(void);
 int32_t Stats_GetSecrets(void);
-void Stats_ShowTotal(const char *filename, GAMEFLOW_LEVEL_TYPE level_type);
+void Stats_ComputeTotal(GAMEFLOW_LEVEL_TYPE level_type, TOTAL_STATS *stats);
 bool Stats_CheckAllSecretsCollected(GAMEFLOW_LEVEL_TYPE level_type);

--- a/src/game/stats.h
+++ b/src/game/stats.h
@@ -11,6 +11,5 @@ void Stats_CalculateStats(void);
 int32_t Stats_GetPickups(void);
 int32_t Stats_GetKillables(void);
 int32_t Stats_GetSecrets(void);
-void Stats_Show(int32_t level_num);
 void Stats_ShowTotal(const char *filename, GAMEFLOW_LEVEL_TYPE level_type);
 bool Stats_CheckAllSecretsCollected(GAMEFLOW_LEVEL_TYPE level_type);

--- a/src/global/types.h
+++ b/src/global/types.h
@@ -1569,12 +1569,12 @@ typedef struct RESUME_INFO {
 } RESUME_INFO;
 
 typedef enum GAME_STATUS {
-    GS_INITIAL = 0,
-    GS_IN_MAIN_MENU = 1,
-    GS_IN_GAME = 2,
-    GS_IN_INVENTORY = 3,
-    GS_IN_PAUSE = 4,
-    GS_IN_STATS = 5,
+    GS_INITIAL,
+    GS_IN_MAIN_MENU,
+    GS_IN_GAME,
+    GS_IN_INVENTORY,
+    GS_IN_PAUSE,
+    GS_IN_STATS,
 } GAME_STATUS;
 
 typedef struct GAME_INFO {

